### DISCUSSION TITLE
Questions about parameter estimation in EM algorithm

### DISCUSSION THREAD

**Discussion by huizizhang949** [theislab/scvelo#800] [Q&A]
Hello,

I have some questions about latent time estimation during EM algorithm. 

Based on equation (4) in the paper, it seems that the model likelihood is invariant to the scaling of the latent time. For example, given the true latent time t and parameter values alpha, beta and gamma, if we increase the latent time t by a factor of, say c, then by decreasing alpha, beta and gamma by a factor of c in the meantime, we can still obtain exactly the same s(t) and u(t) (because in (4), t serves as a multiplication term with either gamma, or beta in the exponent, and alpha, beta and gamma are in the form of devision). Wouldn't this be a problem when using the EM algorithm to find time assignment? How do you fix the scale of the estimated time?

In the tutorial here (https://scvelo.readthedocs.io/DynamicalModeling/, code chunk 9) how to interpret the fitted switching time point? What is the unit of these estimates, hour or minute?
Lastly, I also notice that in the output of code chunk 9, the estimated u0 and s0 (fit_u0 and fit_s0) are always zero for all the genes. Are these the initial values in the induction state? And are they just assumed to be zero (no need to estimate then) or just it happens that the estimates are zero for all genes in this particular example?

Thank you!

Best wishes,
Huizi

**Comment by WeilerP** [comment]
@Huizi949-z, yes, without any restrictions, model parameters and time can only be estimated up to a constant `c`. As part of the `recover_dynamics` function, the estimated time scales are aligned to a common one and the constant `c` is regressed out by fixing the maximal time. This is done in the function `align_dynamics` [here](https://github.com/theislab/scvelo/blob/1805ab4a72d3f34496f0ef246500a159f619d3a2/scvelo/tools/dynamical_model.py#L614).

> In the tutorial here (https://scvelo.readthedocs.io/DynamicalModeling/, code chunk 9) how to interpret the fitted switching time point?

Sorry, not sure what exactly is unclear. It is the time where the system switches from induction to repression.

> What is the unit of these estimates, hour or minute?

There is no unit enforced through the dynamical model. But since it models splicing kinetics which take place over the scale of hours and not minutes, you can assume units to be in hours.

> Lastly, I also notice that in the output of code chunk 9, the estimated u0 and s0 (fit_u0 and fit_s0) are always zero for all the genes. Are these the initial values in the induction state? And are they just assumed to be zero?

Yes.

**Reply by huizizhang949** [reply]
Thanks for the reply!

> As part of the recover_dynamics function, the estimated time scales are aligned to a common one 

Can you please elaborate a bit more on this? What 'common' scale is used and how? For example, in the `align_dynamics` function, what ia done for `mode='align_total_time'`?

> and the constant c is regressed out by fixing the maximal time. 

Sorry I do not quite understand how to 'regress' out the constant c by fixing the maximal time. Is a regression fitted here?

Besdies, is the `fit_alignment_scaling` given in the output of code chunk 9 related to the scaling problem of the time here?

Thank you!
